src/main/java/com/sprint/mission/discodeit/storage/s3/S3BinaryContentStorage.java

```diff
@@ -0,0 +1,108 @@
+package com.sprint.mission.discodeit.storage.s3;
+
+import com.sprint.mission.discodeit.dto.binaryContent.BinaryContentDto;
+import com.sprint.mission.discodeit.storage.BinaryContentStorage;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.time.Duration;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import software.amazon.awssdk.core.ResponseBytes;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.NoSuchKeyException;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+import software.amazon.awssdk.services.s3.presigner.S3Presigner;
+import software.amazon.awssdk.services.s3.presigner.model.GetObjectPresignRequest;
+import software.amazon.awssdk.services.s3.presigner.model.PresignedGetObjectRequest;
+
+@Slf4j
+@Component
+@ConditionalOnProperty(name = "discodeit.storage.type", havingValue = "s3")
+@RequiredArgsConstructor
+public class S3BinaryContentStorage implements BinaryContentStorage {
+
+  private final S3Client s3Client;
+  private final S3Presigner s3Presigner;
+
+  @Value("${discodeit.storage.s3.bucket}")
+  private String bucketName;
+
+  @Value("${discodeit.storage.s3.presigned-url-expiration}")
+  private long presignedUrlExpiration;
+
+  @Override
+  public UUID put(UUID binaryContentId, byte[] bytes) {
+    String key = binaryContentId.toString();
+
+    try {
+      PutObjectRequest request = PutObjectRequest.builder()
+          .bucket(bucketName)
+          .key(key)
+          .build();
+
+      s3Client.putObject(request, RequestBody.fromBytes(bytes));
+      return binaryContentId;
+    } catch (S3Exception e) {
+      throw new RuntimeException("[ERROR] S3 업로드 실패: " + e.awsErrorDetails().errorMessage(), e);
+    }
+  }
+
+  @Override
+  public InputStream get(UUID binaryContentId) {
+    String key = binaryContentId.toString();
+
+    try {
+      GetObjectRequest request = GetObjectRequest.builder()
+          .bucket(bucketName)
+          .key(key)
+          .build();
+
+      ResponseBytes<GetObjectResponse> objectBytes = s3Client.getObjectAsBytes(request);
+      return new ByteArrayInputStream(objectBytes.asByteArray());
```

Comment:
```suggestion
      ResponseInputStream<GetObjectResponse> obj  = s3Client.getObject(request);
      return obj;
```
이렇게 하는건 어때요? 바이트로 읽어서 스트림으로 다시 만드는 것보다 s3에서 제공하는 스트림을 사용해보세요!